config/passport.js

```diff
@@ -0,0 +1,62 @@
+const passport = require( 'passport' )
+const User = require( '../models/users' )
+
+const { Strategy, ExtractJwt } = require( 'passport-jwt' )
+
+const LocalStrategy = require( 'passport-local' )
+
+const localOptions = {
+  usernameField: 'email'
+}
+
+const ERROR_MESSAGE = {
+  error: 'Your login details could not be verified. Please try again.'
+}
+
+const strategy = ( email, password, done ) => {
+  User.findOne({ email }, ( error, user ) => {
+    if(error) {
+      return done( error )
+    }
+
+    if(!user) {
+      return done( null, false, ERROR_MESSAGE )
+    }
+
+    user.comparePassword( password, function( error, isMatch ) {
+      if(error) {
+        return done( error )
+      }
+
+      if(!isMatch) {
+        return done( null, false, ERROR_MESSAGE )
+      }
+
+      return done( null, user )
+    })
+  })
+}
+
+const localLogin = new LocalStrategy( localOptions, strategy )
```

Comment:
Please move these declarations of objects to the top of the module - the flow of the module should be:
* library imports
* imports 
* declarations
* functions
* exports